core/templates/dev/head/pages/community-dashboard-page/community-dashboard-page.controller.ts

```diff
@@ -87,13 +88,50 @@ angular.module('oppia').directive('communityDashboardPage', [
           ctrl.$onInit = function() {
             ctrl.profilePictureDataUrl = null;
             ctrl.username = null;
+            ctrl.userIsReviewer = false;
+            ctrl.userCanReviewTranslationSuggestionsInLanguages = [];
+            ctrl.userCanReviewVoiceoverSuggestionsInLanguages = [];
+            ctrl.userCanReviewQuestions = false;
 
             UserService.getProfileImageDataUrlAsync().then(function(dataUrl) {
               ctrl.profilePictureDataUrl = dataUrl;
             });
 
             UserService.getUserInfoAsync().then(function(userInfo) {
-              ctrl.username = userInfo.getUsername();
+              if (userInfo.isLoggedIn()) {
+                ctrl.username = userInfo.getUsername();
+                UserService.getUserCommunityRightsData().then(
+                  function(userCommunityRights) {
+                    userCommunityRights
+                      .can_review_translation_for_language_codes.forEach(
+                        function(languageCode) {
+                          ctrl.userCanReviewTranslationSuggestionsInLanguages
+                            .push(
+                              LanguageUtilService.getAudioLanguageDescription(
+                                languageCode));
```

Comment:
The deep nesting is difficult to read. Consider extracting one of the functions.